Documentation fixes & improvements (#24546)
idaholab/moose: framework/doc/content/syntax/Bounds/index.md

# Bounds System

> The `Bounds` system is design to bound the value of a nonlinear variable. Whether the bound is an upper or lower

Reviewer: ```suggestion
The `Bounds` system is designed to bound the value of a nonlinear variable. Whether the bound is an upper or lower
```

bound depends on the parameters passed to the `Bounds` object. The bound may be spatially and time-dependent,
even depend on other simulation quantities, as implemented in the particular `Bounds` object used.

The auxiliary variable that serves as the `variable` parameter of a `Bounds` object
is not actually used or even set in the computation. However, its type is used to decide if the `Bounds` loop
will be 'nodal' (loop on all nodes) or 'elemental' (loop on quadrature points in elements). Its block restriction
is used to define where the bounds is applied. It may be re-used for multiple bounds objects.

!alert note

> Only constant elemental variables are supported at this time.

Author: yeah my bad

> Only constant elemental variables are supported at this time.

Author: ```suggestion
Only nodal and constant elemental variables are supported at this time.
```

The `Bounds` system supports both finite element and finite volume variables. Only elemental bounds
should be used for finite volume variables.

Note that in order for `Bounds` to have an effect, the user has to specify the

> PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither options are specified. The PETSc manual pages for the `vinewtonssls` algorithm

Reviewer 2: ```suggestion
PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither option is specified. The PETSc manual pages for the `vinewtonssls` algorithm
```
